transformer_nuggets/llama/finetune.py

```diff
@@ -247,22 +268,26 @@ def train(
                 write_loss_to_file(train_loss_file, step_count, loss_val)
 
                 if rank == 0:
+                    mem_stats = torch.cuda.memory_stats()
+                    peak_active = mem_stats["active_bytes.all.peak"] / (1024 * 1024)
+                    peak_reserved = mem_stats["reserved_bytes.all.peak"] / (1024 * 1024)
                     logging.info(
-                        f"iter={iter_num} max_iters={hyper_params.max_iters} loss={loss_val:.4f}"
+                        f"iter={iter_num} max_iters={hyper_params.max_iters} loss={loss_val:.4f} Peak Active Memory: {peak_active} MB, Peak Reserve Memory: {peak_reserved} MB"
                     )
 
-            if training_config.profile and iter_num < 103:
-                # We want to profile iters 100-102 of the model training
-                p.step()
-
-            if training_config.track_max_memory and rank == 0:
-                logging.info(
-                    "iter_num",
-                    iter_num,
-                    "mem usage GiB",
-                    float(torch.cuda.max_memory_allocated()) / 1024 / 1024 / 1024,
-                )
-            torch.cuda.reset_peak_memory_stats()
+            if training_config.profile:
+                if iter_num < 103:
+                    # dump profile traces in iters 100-102
+                    p.step()
+                if iter_num == 1:
+                    # check memory during both init and train
+                    memory_trace_path = str(
+                        training_config.log_dir
+                        / f"mem_trace_llama_{training_config.model_name}_{dtype_str}_rank_{rank}.pickle"
+                    )
+                    torch.cuda.memory._dump_snapshot(f"{memory_trace_path}")
```

Comment:
I have a context manager for doing memory snapshoting in utils/benchmark, but ... actually as I right this im fine with doing it with this way